src/route.c

```diff
@@ -64,16 +64,18 @@ static timeout_t age_subnets_timeout;
 /* RFC 1071 */
 
 static uint16_t inet_checksum(void *data, int len, uint16_t prevsum) {
-	uint16_t *p = data;
+	uint16_t word;
 	uint32_t checksum = prevsum ^ 0xFFFF;
 
 	while(len >= 2) {
-		checksum += *p++;
+		memcpy(&word, data, sizeof(word));
+		checksum += word;
+		data += 2;
```

Comment:
This fixes the strict aliasing, but now you're doing pointer arithmetic on a `void *`, which is not legal in C or C++, and in fact just a GCC extension. So maybe declare `uint8_t *p = data` and use that?

Reply:
You are right - I have updated the patch now to avoid using a GCC extension.